src/components/SingleArticle.jsx

```diff
@@ -22,6 +43,8 @@ const SingleArticle = () => {
     {isLoading ? <h1>Loading article...</h1> : <section><h1>{selectedArticle.title}</h1>
  <h2>{selectedArticle.author}</h2>
  <p>{selectedArticle.body}</p></section>}
+ <button onClick={()=>handleVoteClick()}>Vote</button><p>Votes: {voteCount}</p>
+ {hasVoted? <button onClick={()=> {undoVote()}}>Undo vote</button>: <p></p>}
```

Comment:
Slight issue with this undoVote functionality, I can upvote however many times I want but can only undo one vote - It would make more sense to allow more undos OR limit the vote to one up vote to match the undo vote functionality (so i only vote up once and i only undo once)